sdk/identity/identity/TOKEN_CACHING.md

```diff
@@ -0,0 +1,76 @@
+
+### In-memory token caching
+
+*In-memory token caching* is the default option provided by the Azure Identity library. This caching approach allows apps to store access tokens in memory. With in-memory token caching, the library first determines if a valid access token for the requested resource is already stored in memory. If a valid token is found, it's returned to the app without the need to make another request to Azure AD. If a valid token is not found, the library will automatically acquire a token by sending a request to Azure AD.
+
+The in-memory token cache provided by the Azure Identity library:
+
+- Is thread-safe.
+- Can be used by multiple threads concurrently.
+
+**Note:** When Azure Identity library credentials are used with Azure service libraries (for example, Azure Blob Storage), the in-memory token caching is active in the `HttpPipeline` layer as well. All `TokenCredential` implementations are supported there, including custom implementations external to the Azure Identity library.
+
+### Persistent token caching
+
+*Persistent disk token caching* is an opt-in feature in the Azure Identity library. The feature allows apps to cache access tokens in an encrypted, persistent storage mechanism. As indicated in the following table, the storage mechanism differs across operating systems.
+
+| Operating system | Storage mechanism |
+|------------------|-------------------|
+| Linux            | Keyring           |
+| macOS            | Keychain          |
+| Windows          | DPAPI              |
+
+With persistent disk token caching enabled, the library first determines if a valid access token for the requested resource is already stored in the persistent cache. If a valid token is found, it's returned to the app without the need to make another request to Azure AD. Additionally, the tokens are preserved across app runs, which:
+
+- Makes the app more resilient to failures.
+- Ensures the app can continue to function during an Azure AD outage or disruption.
+- Avoids having to prompt users to authenticate each time the process is restarted.
+
+#### Code sample
+
+[The sample showcases how to activate persistence token caching](https://github.com/Azure/azure-sdk-for-js/blob/main/sdk/identity/identity/samples/AzureIdentityExamples.md#persist-the-token-cache) in the credentials offered by the Azure Identity library. 
+
+#### Persisting User Authentication Record
+When authenticating a user via `InteractiveBrowserCredential`, `DeviceCodeCredential` or `UsernamePasswordCredential` an `AuthenticationRecord` can be persisted as well. It is returned from the `authenticate` API and contains data identifying an authenticated account. It is needed to identify the appropriate entry in the persisted token cache to silently authenticate on subsequent executions. There is no sensitive data in the `AuthenticationRecord` so it can be persisted in a non-protected state.
```

Comment:
```suggestion
When authenticating a user via `InteractiveBrowserCredential`, `DeviceCodeCredential`, or `UsernamePasswordCredential`, an `AuthenticationRecord` can be persisted as well. It's returned from the `authenticate` API and contains data identifying an authenticated account. It's needed to identify the appropriate entry in the persisted token cache to silently authenticate on subsequent executions. There's no sensitive data in the `AuthenticationRecord`, so it can be persisted in a non-protected state.
```